spec/consensus/proposer-based-timestamp/pbts-sysmodel_002_draft.md

```diff
@@ -0,0 +1,222 @@
+or to the time they retrieve from a trusted source of time, not to a physical clock.
+The restriction imposed by the `PRECISION` parameter can then be used, indirectly,
+to bound the difference block times and real time.
+
+### Message Delays
+
+The assumption that processes have access to synchronized clocks ensures that proposal times
+assigned by *correct processes* have a bounded relation with the real time.
+It is not enough, however, to identify (and reject) proposal times proposed by Byzantine processes.
+
+To properly evaluate whether the time assigned to a proposal is consistent with the real time,
+we need some information regarding the time it takes for a message carrying a proposal
+to reach all its (correct) destinations.
+More precisely, the *maximum delay* for delivering a proposal to its destinations allows
+defining a lower bound, a *minimum time* that a correct process assigns to proposal.
+While *minimum delay* for delivering a proposal to a destination allows defining
+an upper bound, the *maximum time* assigned to a proposal.
+
+#### **[PBTS-MSG-D.0]**
+
+There exists a system parameter `MSGDELAY` for end-to-end delays of messages carrying proposals,
+such for any two correct processes `p` and `q`, and any real time `t`:
+
+- If `p` sends a message `m` carrying a proposal at time `ts`,
+then if `q` receives the message and learns the proposal,
+`q` does that at time `t` such that `ts <= t <= ts + MSGDELAY`.
+
+While we don't want to impose particular restrictions regarding the format of `m`,
+we need to assume that their size is upper bounded.
+In practice, using messages with a fixed-size to carry proposals allows
+for a more accurate estimation of `MSGDELAY`, and therefore is advised.
+
+## Problem Statement
+
+In this section we define the properties of Tendermint consensus
+(cf. the [arXiv paper][arXiv]) in this new system model.
+
+#### **[PBTS-PROPOSE.0]**
+
+A proposer proposes a consensus value `v` with an associated proposal time `v.time`.
+
+#### **[PBTS-INV-AGREEMENT.0]**
+
+[Agreement] No two correct processes decide on different values `v` or proposal times `v.time`.
+
+#### **[PBTS-INV-VALID.0]**
+
+[Validity] If a correct process decides on value `v`,
+then `v` satisfies a predefined `valid` predicate.
+
+#### **[PBTS-INV-TIMELY.0]**
+
+[Time-Validity] If a correct process decides on value `v`,
+then the associated proposal time `v.time` satisfies a predefined `timely` predicate.
+
+> Both [Validity] and [Time-Validity] must be observed even if up to `2f` validators are faulty.
+
+### Timely proposals
+
+The `timely` predicate is evaluated when a process receives a proposal.
+Let `now_p` be time a process `p` reads from its local clock when `p` receives a proposal.
+Let `v` be the proposed value and `v.time` the proposal time.
+The proposal is considered `timely` by `p` if:
+
+1. `now_p >= v.time - PRECISION` and
+1. `now_p <= v.time + MSGDELAY + PRECISION`
+
+### Timely Proof-of-Locks
```

Comment:
If I understand the structure of the document correctly, the following text is about analysis / correctness arguments. If this is right, perhaps we should add a header making this more explicit.

Reply:
A TLA+-related header?

Reply:
Not sure that it is TLA+ related. I guess the point is just that the following text is more interesting when one wants to understand correctness (rather than protocol details). Perhaps we can discuss this synchronously.

Reply:
I've added some headers to the latest version.